src/triggers/commentDocumentV2.ts

```diff
@@ -0,0 +1,254 @@
+  };
+
+  return [comment];
+};
+
+/**
+ * Fetches a list of comments from Linear to use as examples when building a Zap.
+ * @see https://platform.zapier.com/build/cli-hook-trigger#4-write-the-performlist-function
+ * @see https://platform.zapier.com/build/cli-hook-trigger#performlist
+ */
+const getCommentList = () => async (z: ZObject, bundle: Bundle) => {
+  const variables = omitBy(
+    {
+      creatorId: bundle.inputData.creatorId,
+      projectId: bundle.inputData.projectId,
+      documentId: bundle.inputData.documentId,
+    },
+    (v: undefined) => v === undefined
+  );
+
+  const filters = [` { and: [{ documentContent: { null: false } }] }`];
+  if ("creatorId" in variables) {
+    filters.push(`{ user: { id: { eq: $creatorId } } }`);
+  }
+  if ("projectId" in variables) {
+    filters.push(
+      `{ or: [{ documentContent: { project: { id: { eq: $projectId }} } }, { documentContent: { document: { project: { id: { eq: $projectId }}}}}] }`
+    );
+  }
+  if ("documentId" in variables) {
+    filters.push(`{ documentContent: { document: { id: { eq: $documentId }}}}`);
+  }
+
+  const response = await z.request({
+    url: "https://api.linear.app/graphql",
+    headers: {
+      "Content-Type": "application/json",
+      Accept: "application/json",
+      authorization: bundle.authData.api_key,
+    },
+    body: {
+      query: `
+      query ZapierListCommentsV2${
+        Object.keys(variables).length === 0
+          ? ""
+          : `(
+        ${"creatorId" in variables ? "$creatorId: ID" : ""}
+        ${"projectId" in variables ? "$projectId: ID" : ""}
+        ${"documentId" in variables ? "$documentId: ID" : ""}
+      )`
+      } {
+        comments(
+          first: 25
+          ${
+            filters.length > 0
+              ? `
+          filter: {
+            and : [
+              ${filters.join("\n              ")}
+            ]
+          }`
+              : ""
+          }
+        ) {
+          nodes {
+            id
+            body
+            createdAt
+            resolvedAt
+            documentContent {
+              project {
+                id
+                name
+                url
+              }
+              document {
+                id
+                title
+                project {
+                  id
+                  name
+                  url
+                }
+              }
+            }
+            user {
+              id
+              email
+              name
+              avatarUrl
+            }
+            parent {
+              id
+              body
+              createdAt
+              user {
+                id
+                email
+                name
+                avatarUrl
+              }
+            }
+          }
+        }
+      }`,
+      variables: variables,
+    },
+    method: "POST",
+  });
+
+  const data = (response.json as CommentsResponse).data;
+  return data.comments.nodes;
+};
+
+export const newDocumentCommentV2 = {
```

Comment:
I would probably name these something more descriptive, e.g.


```suggestion
export const newDocumentCommentInstant = {
```

or 

```suggestion
export const newDocumentCommentRest = {
```